src/main/java/log/charter/song/CombinedGPBars.java

```diff
@@ -0,0 +1,174 @@
+package log.charter.song;
+
+import java.util.List;
+
+import log.charter.io.gp.gp5.GPMasterBar;
+import log.charter.io.gp.gp5.GPNote;
+import log.charter.io.Logger;
+import log.charter.io.gp.gp5.GPBeat;
+import log.charter.io.gp.gp5.GPBeatEffects;
+import log.charter.io.gp.gp5.GPChord;
+import log.charter.io.gp.gp5.GPDuration;
+import log.charter.util.CollectionUtils.ArrayList2;
+
+public class CombinedGPBars {
+	public class GPBeatUnwrapper extends GPBeat {
+		double note_duration;
+		double note_time_ms;
+
+		public GPBeatUnwrapper(final int tempo, final int dots, final boolean isEmpty, final GPDuration duration,
+				final int tupletNumerator, final int tupletDenominator, final GPBeatEffects beatEffects,
+				final GPChord chord, final List<GPNote> notes, final String text) {
+			super(tempo,dots,isEmpty,duration,tupletNumerator,tupletDenominator,beatEffects,chord,notes,text);
+			this.note_duration = note_duration(gp_duration_to_note_denominator(duration), tupletNumerator, tupletDenominator, dots != 0);
+			this.note_time_ms = note_time_from_duration(tempo);
+		}
+		public GPBeatUnwrapper(final GPBeat beat) {
+			super(beat.tempo,beat.dots,beat.isEmpty,beat.duration,beat.tupletNumerator,beat.tupletDenominator,beat.beatEffects,beat.chord,beat.notes,beat.text);
+			this.note_duration = note_duration(gp_duration_to_note_denominator(duration), tupletNumerator, tupletDenominator, dots != 0);
+			this.note_time_ms = note_time_from_duration(tempo);
+		}
+		
+		final int gp_duration_to_note_denominator(GPDuration duration) {
+			// Example length 16 -> 64/16 = 4 -> 1/4 (quarter)
+			return (64/duration.length);
+		}
+		final double gp_duration_to_time(GPDuration duration) {
+			// Example length 16 -> 64/16 = 4 -> 1/4 (quarter)
+			double note_duration = note_duration(gp_duration_to_note_denominator(duration), this.tupletNumerator, this.tupletDenominator, this.dots != 0);
+			double note_time_ms = note_time_from_custom_duration(this.tempo, note_duration);
+			return note_time_ms;
+		}
+		static final double four_over_four_beat_length(int bpm) {
```

Comment:
rename method according to project conventions, add empty line before, rename variable